### ISSUE TITLE
Problem: Cannot declare actor method in API xml

### ISSUE THREAD

**Issue by sappo** [zeromq/zproject#370]
Currently it is not possible to declare an actor callback method (e.g. used  by zmonitor) in the API xml.

I have a couple of ideas how to solve this issue:
1.
```xml
<method actor = "1">
    My wonderful actor ...
    <argument name = "pipe" type = "zsock" />
    <argument name = "args" type "anything" />
</method>
```
2.
```xml
<constructor actor = "1">
    My wonderful actor ...
</constructor>
```
3.
```xml
<actor>
    My wonderful actor ...
</actor>
```
I'm not sure what is the simplest though as bindings shouldn't wrap this method.

**Comment by hintjens** [comment]
Bindings shouldn't wrap this method; however they could profitably generate an actor wrapper which has methods like send and recv. So the semantics should be "this is an actor." 

We do need to specify the actor handler, so:

```
<actor name = "zmonitor" />
```

Looks right. We might later add methods inside that entity to cover the commands the actor accepts.

**Comment by sappo** [comment]
Okay, just to clarify, did you mean to replace `<class>` with `<actor>` or put `<actor>` inside `<class>`?

**Comment by hintjens** [comment]
Good question. I guess <actor> lives inside <class>, conceptually. It's
like a subclass with a special API that bindings can know about.

**Comment by sappo** [comment]
Sounds about right.
Am 08.12.2015 1:41 nachm. schrieb "Pieter Hintjens" <
notifications@github.com>:

**Comment by sappo** [comment]
solved by #386
